### ISSUE TITLE
Enhanced PYKE

### ISSUE THREAD

**Issue by Demirrr** [dice-group/DAIKIRI-Embedding#5]
If time allows, we could implement PYKE in pytorch and enhance it by following

1. Explore other similarity metrics, e.g. counstruct Laplacian matrix.
2. Use torch.nn.MarginRankingLoss 
3. Use ADAM as optimizer. Note that as as the learning rate decays, the rate of change in the embeddings will decrease, i.e. the total energy in the system diminishes.

**Comment by Demirrr** [comment]
I think this idea can be considered a possible topic for thesis.
